programs/keeper-client/KeeperClient.cpp

```diff
@@ -311,8 +324,31 @@ int KeeperClient::main(const std::vector<String> & /* args */)
         return 0;
     }
 
-    auto host = config().getString("host", "localhost");
-    auto port = config().getString("port", "9181");
+    DB::ConfigProcessor config_processor(config().getString("config-file", "config.xml"));
+
+    /// This will handle a situation when clickhouse is running on the embedded config, but config.d folder is also present.
+    config_processor.registerEmbeddedConfig("config.xml", "<clickhouse/>");
+    auto clickhouse_config = config_processor.loadConfig();
+
+    String host;
+    String port;
+    String prefix = "zookeeper.node[0].";
```

Comment:
If I understand correctly, you add only the first connection string here, why wouldn't you add them all and let the client connect to one of them?